### DISCUSSION TITLE
What are the minimum system requirements to install and run zitadel locally and in production?

### DISCUSSION THREAD

**Discussion by gc-ss** [zitadel/zitadel#2079] [Q&A]
What are the minimum system requirements to:

1. install/build the zitadel containers
2. run zitadel locally for test/customer software dev purpose (not developing zitadel itself)
3. run zitadel in production?

**Comment by gc-ss** [comment]
Since I'm currently in the install/build the zitadel containers stage, I was able to build the zitadel containers using docker-compose on an AWS `i3en.large` with the SSD mounted.

It's possible the install/build will work on smaller instances but the CPU and RAM pressure was pretty high on the `i3en.large` instance throughout the install/build.

Time taken was ~14 minutes

**Reply by fforootd** [reply]
## Production HA Cluster

A minimal HA Cluster with ZITADEL normally is built around 3 Nodes with 4CPU and 16GB Memory each.
However they also contain  things like log collection, metrics, Grafana and so on.

So this could be reduced to around 4CPU and 8GB Ram each, when leaving that out.

## General Resource Usage

ZITADEL consumes around 512MB Ram and can easy run with less then 1CPU core. But beware of CPU spike when hashing passwords. We recommend to have 4CPU cores available for this purpose. The database consumes around 2 CPU und normal conditions and 6GB Ram with some caching to it.

## Container Images

We recommend using our [prebuilt images](https://github.com/orgs/caos/packages?repo_name=zitadel). So that you don't need to build when running ZITADEL
